### PR TITLE
chore: engine api & evm activation

### PR DESCRIPTION
**Pull request by danyalprout**
### Description
- Add the new engine API V5 call
- Ensure EVM activates Base V1 spec
- Add action test for chain crossing the boundary

### CODE REVIEW

File: actions/harness/tests/base_v1_activation.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    let l1_chain = SharedL1Chain::from_blocks(h.l1.chain().to_vec());
+    let mut builder = h.create_l2_sequencer(l1_chain);
+
+    for _ in 1..=4u64 {
+        let block = builder.build_next_block().expect("build L2 block");
+        let mut source = ActionL2Source::new();
+        source.push(block);
+        let mut batcher = Batcher::new(source, &h.rollup_config, batcher_cfg.clone());
+        batcher.advance(&mut h.l1).await.expect("advance");
```
Comment (refcell) [inline actions/harness/tests/base_v1_activation.rs]: ugh need to fix this in other tests as well, we shouldn't be creating a new batcher inside block publishing loops.... this is probably because in other tests I still have this pattern. Instead, we should persist the same batcher across the test and just use it here to post the block.

File: actions/harness/tests/base_v1_activation.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    let (mut verifier, _chain) = h.create_verifier_from_sequencer(
+        &builder,
+        SharedL1Chain::from_blocks(h.l1.chain().to_vec()),
+    );
+    verifier.initialize().await.expect("initialize");
```
Comment (refcell) [inline actions/harness/tests/base_v1_activation.rs]: I need to get rid of these expects at the call sites. Since these are tests we should just inline the assertions that these construct successfully.

File: crates/alloy/rpc-types-engine/src/payload/mod.rs [hunk 3]
```diff
@@ -7,16 +7,21 @@ mod v3;
...
 use alloy_consensus::{Block, BlockHeader, HeaderInfo, Transaction};
 use alloy_eips::{Decodable2718, Encodable2718, Typed2718, eip7685::EMPTY_REQUESTS_HASH};
 use alloy_primitives::{Address, B256, Bytes, Sealable, U256};
+pub use alloy_rpc_types_engine::BlobsBundleV2;
 use alloy_rpc_types_engine::{
     ExecutionPayload, ExecutionPayloadInputV2, ExecutionPayloadV1, ExecutionPayloadV2,
     ExecutionPayloadV3, PayloadError,
 };
-pub use v4::{OpExecutionPayloadEnvelopeV4, OpExecutionPayloadV4};
+pub use v5::OpExecutionPayloadEnvelopeV5;
```
Comment (refcell) [inline crates/alloy/rpc-types-engine/src/payload/mod.rs]: nit this should be placed under the `mod v5;` line.

File: crates/alloy/rpc-types-engine/src/payload/v5.rs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+/// This structure maps for the return value of `engine_getPayload` of the beacon chain spec, for
+/// V5.
+///
+/// The OP variant follows the same pattern as V4: replaces `ExecutionPayloadV3` with
+/// [`OpExecutionPayloadV4`] (which adds `withdrawalsRoot`), and keeps all other fields identical
+/// to the mainnet [`ExecutionPayloadEnvelopeV5`](alloy_rpc_types_engine::ExecutionPayloadEnvelopeV5).
+///
+/// See also:
+/// [execution payload envelope v5] <https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/exec-engine.md#engine_getpayloadv5>
```
Comment (refcell) [inline crates/alloy/rpc-types-engine/src/payload/v5.rs]: We should probably point to our specs in `docs/specs/`

File: crates/execution/rpc/src/engine.rs [hunk 5]
```diff
@@ -29,6 +29,7 @@ pub const OP_ENGINE_CAPABILITIES: &[&str] = &[
     "engine_getPayloadV2",
     "engine_getPayloadV3",
     "engine_getPayloadV4",
+    "engine_getPayloadV5",
     "engine_newPayloadV2",
     "engine_newPayloadV3",
     "engine_newPayloadV4",
```
Comment (refcell) [inline crates/execution/rpc/src/engine.rs:35]: shouldn't `engine_newPayloadV5` also be added here?

**Review by refcell — COMMENTED** [review]
Commets

File: crates/execution/rpc/src/engine.rs (hunk 5 above)
Comment (danyalprout) [reply crates/execution/rpc/src/engine.rs:35]: `engine_newPayloadV5` is only for Amsterman, unfortunately these API's don't actually seem to be paired together. 

https://github.com/paradigmxyz/reth/blob/6736b2ad659d49e6951b3dd88b55f6a89f32a999/crates/rpc/rpc-api/src/engine.rs#L77

https://github.com/paradigmxyz/reth/blob/6736b2ad659d49e6951b3dd88b55f6a89f32a999/crates/rpc/rpc-api/src/engine.rs#L183

File: crates/alloy/rpc-types-engine/src/payload/v5.rs (hunk 4 above)
Comment (danyalprout) [reply crates/alloy/rpc-types-engine/src/payload/v5.rs]: Agreed

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by refcell — APPROVED** [review]
